python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -697,39 +852,145 @@ def setDefaults(self):
+    \endcode
+    This will process the HSC-I band data. The results are written in $CI_HSC_DIR/DATA/deepCoadd-results/HSC-I 
+    You may also choose to run:
+    \code
+    scons warp-903334 warp-903336 warp-903338 warp-903342 warp-903344 warp-903346
+    assembleCoadd.py $CI_HSC_DIR/DATA --id patch=5,4 tract=0 filter=HSC-R --selectId visit=903334 ccd=16 --selectId visit=903334 ccd=22 --selectId visit=903334 ccd=23 --selectId visit=903334 ccd=100 --selectId visit=903336 ccd=17 --selectId visit=903336 ccd=24 --selectId visit=903338 ccd=18 --selectId visit=903338 ccd=25 --selectId visit=903342 ccd=4 --selectId visit=903342 ccd=10 --selectId visit=903342 ccd=100 --selectId visit=903344 ccd=0 --selectId visit=903344 ccd=5 --selectId visit=903344 ccd=11 --selectId visit=903346 ccd=1 --selectId visit=903346 ccd=6 --selectId visit=903346 ccd=12
+    \endcode
+    to generate the coadd for the HSC-R band if you are interested in following multiBand Coadd processing as
+    discussed in \ref pipeTasks_multiBand.
     """
     ConfigClass = SafeClipAssembleCoaddConfig
     _DefaultName = "safeClipAssembleCoadd"
 
     def __init__(self, *args, **kwargs):
+        """!
+        \brief Initialize the task and make the \ref SourceDetectionTask_ "clipDetection" subtask.
+        """
         AssembleCoaddTask.__init__(self, *args, **kwargs)
         schema = afwTable.SourceTable.makeMinimalSchema()
         self.makeSubtask("clipDetection", schema=schema)
 
     def assemble(self, skyInfo, tempExpRefList, imageScalerList, weightList, bgModelList, *args, **kwargs):
-        """Assemble the coadd for a region
+        """!
+        \brief Assemble the coadd for a region
+
+        Compute the difference of coadds created with and without outlier rejection to identify coadd pixels
+        that have outlier values in some individual visits. Detect clipped regions on the difference image and
+        mark these regions on the one or two individual coaddTempExps where they occur if there is significant
+        overlap between the clipped region and a source.
+        This leaves us with a set of footprints from the difference image that have been identified as having
+        occured on just one or two individual visits. However, these footprints were generated from a
+        differece image. It is conceivable for a large diffuse source to have become broken up into multiple
+        footprints acrosss the coadd difference in this process.
+        Redo the clip detection by determining the clipped region from all overlapping footprints from the
+        detected sources in each visit - these are big footprints.
```

Comment:
It is not quite correct that you redo the clip detection, but rather try to merge existing clipped detections.  You also only look at the large footprints from each visit not all of them in this step.